classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -9,6 +9,13 @@
  */
 class ActionScheduler_DBStore extends ActionScheduler_Store {
 
+	/**
+	 * Used to share information about the before_date property of claims internally.
+	 *
+	 * @var array
+	 */
+	private $before_dates_by_claim_id = array();
```

Comment:
:memo: I'm using a private method to share this information rather than modify any of the method signatures (this is for backwards compatibility reasons...if we altered some of the methods to add a new optional param, and passed the information that way, then under PHP 8.0+ we'd face errors if third-party sub-classes didn't update in time).